5.4.rational_Harry.py

```diff
@@ -0,0 +1,38 @@
+# הארי רציונלי אבל לא נורא
+
+# install if not exist "bs4" need for html parser
+
+from bs4 import BeautifulSoup
+import re
+import os
+import bs4
+
+def change_name_in_folder_by_func(get_name_func , directory=None):
+    # iterate over files in
+    # that directory
+    for filename in os.scandir(directory):
```

Comment:
You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.